kaggle/api/kaggle_api_extended.py

```diff
@@ -86,98 +86,248 @@ class KaggleApi(KaggleApi):
+                sys.exit('The specified proxy ' + 
+                         config_data[self.CONFIG_NAME_PROXY] +
+                         ' is not valid, please check your proxy settings')
             else:
-                raise ValueError(
-                    'Unauthorized: you must download an API key from '
-                    'https://www.kaggle.com/<username>/account\nThen put ' +
-                    self.config_file + ' in the folder ' + self.config_dir)
+                sys.exit('Unauthorized: you must download an API key or export '
+                         'credentials to the environment. Please see\n ' +
+                         'https://github.com/Kaggle/kaggle-api#api-credentials '
+                          + 'for instructions.')
 
-    def set_config_value(self, name, value, quiet=False):
+
+    def _authenticate_config(self, quiet=False):
+        '''autheticate config is the first effort to get a username
+           and key to authenticate to the Kaggle API. Since we can get the
+           username and password from the environment, it's not required.
+
+           Parameters
+           ==========
+           configuration: the Configuration object to save a username and
+                          password, if defined
+           quiet: add verbosity
+        '''
+        config_data = {}
+
+        if os.path.exists(self.config):
+
+            try:
+                if os.name != 'nt':
+                    permissions = os.stat(self.config).st_mode
+                    if (permissions & 4) or (permissions & 32):
+                        print('Warning: Your Kaggle API key is readable by other' 
+                              'users on this system! To fix this, you can run' + 
+                              '\'chmod 600 {}\''.format(self.config))
+
+                with open(self.config, 'r') as f:
+                    config_data = json.load(f)
+            except:
+                pass
+
+        else:
+
+            # Warn the user that configuration will be reliant on environment
+            if not quiet:
+                print('No Kaggle API config file found, will use environment.')
+
+        return config_data
+
+  
+    def _read_config(self):
+        '''read in the configuration file, a json file defined at self.confi'''
+        
         try:
             with open(self.config, 'r') as f:
                 config_data = json.load(f)
-            if value is not None:
-                config_data[name] = value
-                with open(self.config, 'w') as f:
-                    json.dump(config_data, f, indent=2)
-            if name in config_data:
-                self.config_values[name] = config_data[name]
-        except:
-            pass
-        if not quiet:
-            self.print_config_value(name, separator=' is now set to: ')
+        except FileNotFoundError:
+            config_data = {}
+
+        return config_data
+
+
+    def _write_config(self, config_data, indent=1):
```

Comment:
Keep `indent=2` please (unless there's a reason that I'm unaware of for it to be 1?)

Reply:
Which indent are you referring to?

Reply:
The argument for the method

Reply:
Oh geez I'm really dense, the actual argument (I was counting spaces).